### PR TITLE
Remove code related to old single kernel picker

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/test/datascience/notebook/remoteNotebookEditor.vscode.test.ts [hunk 1]
```diff
@@ -161,44 +161,11 @@ suite('Remote Kernel Execution', function () {
                 )}`
         );
     });
-
-    test('Remote kernels are removed when switching to local @kernelPicker', async function () {
-        await controllerLoader.loaded;
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/remoteNotebookEditor.vscode.test.ts:166]: This is not possible with the new and stable kernel pickers, this applies to the old kernel picker

File: src/test/datascience/notebook/remoteNotebookEditor.vscode.test.ts [hunk 2]
```diff
@@ -161,44 +161,11 @@ suite('Remote Kernel Execution', function () {
...
-        );
-    });
-
     test('Local Kernel state is not lost when connecting to remote @kernelPicker', async function () {
         await controllerLoader.loaded;
 
         // After resetting connection to local only, verify all remote connections are no longer available.
         await jupyterServerSelector.setJupyterURIToLocal();
-        await waitForCondition(
-            async () => {
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/remoteNotebookEditor.vscode.test.ts:195]: this applies to the old kernel picker

File: src/notebooks/controllers/controllerLoader.ts [hunk 3]
```diff
@@ -128,20 +128,6 @@ export class ControllerLoader implements IControllerLoader, IExtensionSyncActiva
                     return connection.id === controller.connection.id;
                 });
 
-                if (this.featuresManager.features.kernelPickerType !== 'Insiders') {
```
Comment (DonJayamanne) [inline src/notebooks/controllers/controllerLoader.ts:131]: this applies to the old kernel picker

File: src/kernels/raw/finder/contributedLocalKernelSpecFinder.node.ts [hunk 4]
```diff
@@ -169,32 +168,25 @@ export class ContributedLocalKernelSpecFinder
     }
 
     public get kernels(): LocalKernelConnectionMetadata[] {
-        if (this.featureManager.features.kernelPickerType === 'Insiders') {
```
Comment (DonJayamanne) [inline src/kernels/raw/finder/contributedLocalKernelSpecFinder.node.ts:172]: kernels will be found the same regardless of stable or insider

File: src/kernels/kernelFinder.ts [hunk 5]
```diff
@@ -79,51 +76,46 @@ export class KernelFinder implements IKernelFinder {
 
         // List kernels might be called after finders or connections are removed so just clear out and regenerate
         this.connectionFinderMapping.clear();
-        let finders: IContributedKernelFinder<KernelConnectionMetadata>[] = [];
```
Comment (DonJayamanne) [inline src/kernels/kernelFinder.ts:82]: removed t!his if condition

**Review by rebornix — APPROVED** [review]

**Pull request merged**